common/cuda_hip/matrix/batch_dense_kernels.hpp.inc

```diff
@@ -170,3 +170,74 @@ __global__ void scale_kernel(
               mat_b.values);
     }
 }
+
+
+template <typename ValueType>
+__device__ __forceinline__ void scale_add(
+    const ValueType alpha,
+    const gko::batch::matrix::dense::batch_item<const ValueType>& mat,
+    const gko::batch::matrix::dense::batch_item<ValueType>& in_out)
+{
+    for (int iz = threadIdx.x; iz < mat.num_rows * mat.num_cols;
+         iz += blockDim.x) {
+        const int row = iz / mat.stride;
+        const int col = iz % mat.stride;
+        in_out.values[row * in_out.stride + col] =
+            alpha * in_out.values[row * in_out.stride + col] +
+            mat.values[row * mat.stride + col];
```

Comment:
```suggestion
        const int row = iz / mat.num_cols;
        const int col = iz % mat.num_cols;
        in_out.values[row * in_out.stride + col] =
            alpha * in_out.values[row * in_out.stride + col] +
            mat.values[row * mat.stride + col];
```
if you would like to use stride, need to ensure col is less than mat.num_cols and `iz < mat.num_rows * mat.stride`
I think you do not support stride > num_cols, so maybe add the todo note for tracking at least